cpp/ql/src/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -268,6 +268,32 @@ class IRGuardCondition extends Instruction {
           ne.controls(controlled, testIsTrue.booleanNot())) 
     }
 
+    /**
+     * Holds if `branch` jumps directly to `succ` when this condition is `testIsTrue`.
+     * 
+     * This predicate is intended to help with situations in which an inference can only be made
+     * based on an edge between a block with multiple successors and a block with multiple
+     * predecessors. For example, in the following situation, an inference can be made about the
+     * value of `x` at the end of the `if` statement, but there is no block which is controlled by
+     * the `if` statement when `x >= y`.
+     * ```
+     * if (x < y) {
+     *   x = y;
+     * }
+     * return x;
+     * ```
+     */
+    predicate controlsEdgeDirectly(ConditionalBranchInstruction branch, IRBlock succ, boolean testIsTrue) {
```

Comment:
I still think this is a misleading name.  I'd go with something like `hasBranchEdge`.  To me, "guard controls X" means that a certain branch edge from the guard dominates X in the sense that it must have been taken on all valid paths leading to X; this can then either manifest as "controls directly" when the dominance can be determined directly from the dominance graph, or more indirectly as in the following where `guard` being true controls X, without this being apparent from a naive dominance calculation, but needs a sequence of more complicated logical implications:
```
b = guard && foo;
if (b) { X }
```

Reply:
Alright, changed in next push